### PR TITLE
Comply with guide for method doc: array.c

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Methods considered:
- delete_at
- slice!
- reject!
- reject
- delete_if
- zip
- transpose
- replace
- clear
- fill

### CODE REVIEW

File: array.c [hunk 1]
```diff
@@ -4466,29 +4410,21 @@ rb_ary_clear(VALUE ary)
 /*
  *  call-seq:
  *    array.fill(obj) -> self
- *    array.fill(obj, start) -> self
- *    array.fill(obj, start, length) -> self
+ *    array.fill(obj, start, length = self.size) -> self
```
Comment (jeremyevans) [inline array.c:4413]: I don't think these cases are the same, so they should use a separate call-seq (as before), and documented separately.

```ruby
a = ['a', 'b', 'c', 'd']
a.dup.fill(:X, 2)
=> ["a", "b", :X, :X]
a.dup.fill(:X, 2, a.size)
=> ["a", "b", :X, :X, :X, :X]
```

Comment (BurdetteLamar) [reply array.c:4413]: call-seq restored.  Do the examples need to change?

**Comment by BurdetteLamar** [comment]
Thanks.  I'd prefer to clean this up before merging, so will look at it again tomorrow.

**Comment by BurdetteLamar** [comment]
I still don't get it.  I'll go ahead and merge the compliance part.

**Pull request merged**